pallets/pallet-did-lookup/src/signature.rs

```diff
@@ -16,18 +16,34 @@
 
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 
+use codec::alloc::string::ToString;
 use sp_std::vec::Vec;
 
 // According to https://github.com/polkadot-js/common/blob/5d5c7e4c0ace06e3301ccadfd3c3351955f1e251/packages/util/src/u8a/wrap.ts#L13
-
 const PAYLOAD_BYTES_WRAPPER_PREFIX: &[u8; 7] = b"<Bytes>";
 const PAYLOAD_BYTES_WRAPPER_POSTFIX: &[u8; 8] = b"</Bytes>";
+const ETHEREUM_SIGNATURE_PREFIX: &[u8; 26] = b"\x19Ethereum Signed Message:\n";
+pub enum WrapType {
+	Substrate,
+	Ethereum,
+}
+pub trait GetWrapType {
+	fn get_wrap_type(&self) -> WrapType;
+}
 
-pub(crate) fn get_wrapped_payload(payload: &[u8]) -> Vec<u8> {
-	PAYLOAD_BYTES_WRAPPER_PREFIX
-		.iter()
-		.chain(payload.iter())
-		.chain(PAYLOAD_BYTES_WRAPPER_POSTFIX.iter())
-		.copied()
-		.collect()
+pub(crate) fn get_wrapped_payload<A: GetWrapType>(payload: &[u8], account: &A) -> Vec<u8> {
+	match account.get_wrap_type() {
+		WrapType::Substrate => PAYLOAD_BYTES_WRAPPER_PREFIX
+			.iter()
+			.chain(payload.iter())
+			.chain(PAYLOAD_BYTES_WRAPPER_POSTFIX.iter())
+			.copied()
+			.collect(),
+		WrapType::Ethereum => ETHEREUM_SIGNATURE_PREFIX
+			.iter()
+			.chain(payload.len().to_string().as_bytes().iter())
```

Comment:
Can you add a comment that the length of the payload is also added to the message to sign?